analyzers/src/SonarAnalyzer.CSharp/Extensions/LocalFunctionStatementSyntaxWrapperExtensions.cs

```diff
@@ -0,0 +1,30 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2022 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using StyleCop.Analyzers.Lightup;
+
+namespace SonarAnalyzer.Extensions;
+
+public static class LocalFunctionStatementSyntaxWrapperExtensions
+{
+    public static bool IsTopLevel(this LocalFunctionStatementSyntaxWrapper localFunction) =>
+        // The tree looks like: CompilationUnit -> GlobalStatement -> LocalFunction and we check the compilation unit to be top-level main
+        localFunction.SyntaxNode.Parent.Parent.IsTopLevelMain();
```

Comment:
This may throw NREs and you can get rid of the comment, by just testing for it. This also removes the `IsTopLevelMain`:
```suggestion
        localFunction is { SyntaxNode: { Parent: GlobalStatementSyntax { Parent: CompilationUnitSyntax } } };
```